apps/builder/app/shared/nano-states/components.ts

```diff
@@ -5,11 +5,51 @@ import {
   type AnyComponent,
   type WsComponentMeta,
   type WsComponentPropsMeta,
+  type Hook,
+  type HookContext,
 } from "@webstudio-is/react-sdk";
 import type { Instance } from "@webstudio-is/project-build";
+import { subscribe } from "~/shared/pubsub";
+import { instancesStore } from "./instances";
+import { dataSourceVariablesStore, propsStore } from "./nano-states";
+
+type HookCommand = NonNullable<
+  {
+    [Name in keyof Hook]: {
+      name: Name;
+      data: Parameters<NonNullable<Hook[Name]>>[1];
+    };
+  }[keyof Hook]
+>;
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    emitComponentHook: HookCommand;
+  }
+}
+
+export const subscribeComponentHooks = () => {
```

Comment:
interresting, so we have now a general way to emit events that are 1:1 mapped to hooks? or do I misunderstand something?

maybe worth a comment for what it is with an example of a problem its solving